src/plugins/rehype-external-link.ts

```diff
@@ -0,0 +1,77 @@
+import { visit } from "unist-util-visit";
+export default rehypeExternalLink;
+/**
+ * Rehype plugin: automatically adds the Starlight-style external link icon
+ * to all external <a> tags in Markdown content.
+ */
+function rehypeExternalLink() {
+  const iconNode = getExternalLinkSvgNode();
+  let isTagaInsideH2 = false;
```

Comment:
I believe this would read better:

```suggestion
  let isATagInsideH2 = false;
```